flutter_local_notifications/windows/flutter_local_notifications_plugin.cpp

```diff
@@ -0,0 +1,308 @@
+			}
+		}
+		else if (method_name == Method::CANCEL && toastNotifier.has_value()) {
+			const auto args = std::get_if<flutter::EncodableMap>(method_call.arguments());
+			if (args != nullptr) {
+				const auto id = Utils::GetMapValue<int>("id", args).value();
+				const auto group = Utils::GetMapValue<std::string>("group", args);
+
+				CancelNotification(id, group);
+				result->Success();
+			}
+			else {
+				result->Error("INTERNAL", "flutter_local_notifications encountered an internal error.");
+			}
+		}
+		else if (method_name == Method::CANCEL_ALL && toastNotifier.has_value()) {
+			CancelAllNotifications();
+			result->Success();
+		}
+		else {
+			result->NotImplemented();
+		}
+	}
+
+	std::optional<bool> FlutterLocalNotificationsPlugin::HasIdentity() {
+		if (!IsWindows8OrGreater()) {
+			// OS is windows 7 or lower
+			return false;
+		}
+
+		UINT32 length;
+		auto err = GetCurrentPackageFullName(&length, nullptr);
+		if (err != ERROR_INSUFFICIENT_BUFFER) {
+			if (err == APPMODEL_ERROR_NO_PACKAGE)
+				return false;
+
+			return std::nullopt;
+		}
+
+		PWSTR fullName = (PWSTR)malloc(length * sizeof(*fullName));
+		if (fullName == nullptr)
+			return std::nullopt;
+
+		err = GetCurrentPackageFullName(&length, fullName);
+		if (err != ERROR_SUCCESS)
+			return std::nullopt;
+
+		free(fullName);
+
+		return true;
+	}
+
+	bool FlutterLocalNotificationsPlugin::Initialize(
+		const std::string& appName,
+		const std::string& aumid,
+		const std::optional<std::string>& iconPath,
+		const std::optional<std::string>& iconBgColor
+	) {
+		_aumid = winrt::to_hstring(aumid);
+		PluginRegistration::RegisterApp(aumid, appName, iconPath, iconBgColor, channel);
+		
+		const auto hasIdentity = HasIdentity();
+		if (!hasIdentity.has_value())
+			return false;
+
+		if (hasIdentity.value())
+			toastNotifier = winrt::Windows::UI::Notifications::ToastNotificationManager::CreateToastNotifier();
+		else
+			toastNotifier = winrt::Windows::UI::Notifications::ToastNotificationManager::CreateToastNotifier(winrt::to_hstring(aumid));
+
+		return true;
+	}
+
+	void FlutterLocalNotificationsPlugin::ShowNotification(
+		const int id,
+		const std::optional<std::string>& title,
+		const std::optional<std::string>& body,
+		const std::optional<std::string>& payload,
+		const std::optional<std::string>& group
+	) {
+		// obtain a notification template with a title and a body
+		//const auto doc = winrt::Windows::UI::Notifications::ToastNotificationManager::GetTemplateContent(winrt::Windows::UI::Notifications::ToastTemplateType::ToastText02);
+		// find all <text /> tags
+		//const auto nodes = doc.GetElementsByTagName(L"text");
+
+		XmlDocument doc;
+		doc.LoadXml(L"\
+			<toast>\
+				<visual>\
+					<binding template=\"ToastGeneric\">\
+					</binding>\
+				</visual>\
+			</toast>");
+
+		const auto bindingNode = doc.SelectSingleNode(L"//binding[1]");
+
+		if (title.has_value()) {
```

Comment:
According to this link: https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/adaptive-interactive-toasts?tabs=xml
Maybe we can pass the XML text directly to the plugin and then combine the title and content etc into XML text on the Dart code side?
In this way, more Toast styles can be easily supported in the future.

Reply:
That is something that I have thought about actually - a Windows specific method that allows users to pass in XML of the notification toast directly.